exchanges/collateral/collateral.go

```diff
@@ -0,0 +1,71 @@
+package collateral
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// Valid returns whether the collateral mode is valid
+func (t Mode) Valid() bool {
+	return t != UnsetMode && supportedCollateralModes&t == t
+}
+
+// UnmarshalJSON converts json into collateral mode
+func (t *Mode) UnmarshalJSON(d []byte) error {
+	var mode string
+	err := json.Unmarshal(d, &mode)
+	if err != nil {
+		return err
+	}
+	*t, err = StringToMode(mode)
+	return err
+}
+
+// String returns the string representation of the collateral mode in lowercase
+// the absence of a lower func should hopefully highlight that String is lower
+func (t Mode) String() string {
+	switch t {
+	case UnsetMode:
+		return unsetCollateralStr
+	case SingleMode:
+		return singleCollateralStr
+	case MultiMode:
+		return multiCollateralStr
+	case GlobalMode:
```

Comment:
```suggestion
	case PortfolioMode:
```

Portfolio margin is typically used by exchanges if the margin type is applied to the entire account, so I think this would be a more accurate name (and across this PR) to use